presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionTreeUtils.java

```diff
@@ -112,4 +122,33 @@ public static boolean isEqualComparisonExpression(Expression expression)
     {
         return expression instanceof ComparisonExpression && ((ComparisonExpression) expression).getOperator() == ComparisonExpression.Operator.EQUAL;
     }
+
+    public static Optional<EnumType> maybeGetEnumTypeFromLiteral(DereferenceExpression node, TypeManager typeManager)
+    {
+        try {
+            Type baseType = typeManager.getType(parseTypeSignature(node.getBase().toString()));
+            if (baseType instanceof EnumType) {
+                return Optional.of((EnumType) baseType);
+            }
+        }
+        catch (IllegalArgumentException e) {
+            return Optional.empty();
+        }
+        return Optional.empty();
+    }
+
+    public static boolean isEnumLiteral(DereferenceExpression node, Type nodeType)
+    {
+        return nodeType instanceof EnumType && node.getBase().toString().equalsIgnoreCase(nodeType.toString());
```

Comment:
These functions ideally should all just take `QualifiedName` if `DereferenceExpression` is used, either get QualifiedName from it, or you need to check the base is an `Identifier`. Blindly use `toString` is error prune.

Reply:
I'm keeping the signature as-is because most of the callsites would have to extract the QualifiedName otherwise, but I'm following your suggestion to parse the QualifiedName inside the method.